gridftp/server/src/test/sharing_allowed_test.c

```diff
@@ -137,19 +138,30 @@ int main()
                 strerror(errno));
         exit(99);
     }
-    if (rc < 2)
+
+    grent = NULL;
+    rc = getgroups(NGROUPS_MAX, my_groups);
```

Comment:
Perhaps good to also fail on `rc<0`

Reply:
The result from getgroups() is not checked for negative values anywhere in the code.
See: gridftp/server/src/globus_i_gfs_data.c

If -1 is returned in the test code, the loop will be skipped, skip_other_group will be set to true and the secondary group tests will be skipped.

Reply:
Actually, I think the code was written so that getgroups can not return -1. According to the man page, there are two error conditions:

       EFAULT list has an invalid address.
       EINVAL size is less than the number of supplementary group IDs, but is not zero.

These error conditions are checked in the code before calling getgroups.

Reply:
That's indeed true for Linux (and MacOS if we can believe the man page), but not per se for other OSes. We can leave it as is, but personally I find it good practice to check for errors even if with the current code, they cannot appear.